pkg/controller/workspace/prerequisites/generate_prerequisites.go

```diff
@@ -58,19 +58,6 @@ func generatePrerequisites(namespace string) ([]runtime.Object, error) {
 				},
 			},
 		},
-		&rbacv1.Role{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "view-workspaces",
```

Comment:
Reading Workspace CR directly has its own benefit: Rest API Sidecar always gets the latest changes. With this approach Theia won't be restarted if command is updated, and Rest API Sidecar won't get the latest changes.

So, in general I would leave reading Workspace CR but try to avoid using Runtime additional field in the WorkspaceStatus, instead I would investigate the possible model for WorkspaceStatus to cover Rest API needs and make sidecar able to generate Runtime for itself based on the status. Let me imagine:
```yaml
kind: Workspace
spec:
  started: true
  devfile:
    components:
      - type: cheEditor
        alias: theia
        id: eclipse/che-theia/nightly
      - type: dockerimage
        alias: maven
        dockerimage: maven:3.10.0
    commands:
      - name: build
        componenent: maven
        exec: mvn clean install
status:
  editorURL: "https://workspace123-theia.apps-testing.com"
  endpoints:
    - name: ide
      component: theia
      url: https://workspace123-theia.apps-testing.com
      status: Exposing
      type: ide
    - name: webviews
      component: theia
      url: https://workspace123-webviews.apps-testing.com
      status: Unknown
 containers:
    - name: theia
      status: Running
      image: eclipse/che-theia:next
      component: theia
       #? podName: ?
    - component: maven
      image: maven:3.10.0
      resources:
        limit:
          memory: 1Gi
          cpu: 2
#^ commands are not in the status since Sidecar is able to match commands from spec to containers from status using component field.
```

Reply:
Mounting a configmap in the way we do correctly propagates changes to che-rest-apis:

![update-workspace-propagation](https://user-images.githubusercontent.com/16168279/79353728-76a24280-7f09-11ea-9f89-c2c368004391.gif)